scipy/signal/spectral.py

```diff
@@ -402,7 +403,7 @@ def csd(x, y, fs=1.0, window='hann', nperseg=256, noverlap=None, nfft=None,
 
 def spectrogram(x, fs=1.0, window=('tukey',.25), nperseg=256, noverlap=None,
                 nfft=None, detrend='constant', return_onesided=True,
-                scaling='density', axis=-1, mode='psd'):
+                scaling='density', axis=-1):
```

Comment:
Good point. What started me down this path was gh-5757, where the complex mode was not being calculated correctly. I later realized that the usual name for a complex spectrogram is the STFT, leading to these function. I suppose one way to handle the backwards compatibility would be to alias `mode='complex'` to `stft`, and maybe even emit a warning that the alias is happening...